frontend/src/components/nav/navbar.js

```diff
@@ -56,6 +57,15 @@ class NavBar extends React.Component {
         }
     }
 
+    debounceSearch(func, delay) {
+        let inDebounce;
+        return (e) => {
+            e.persist();
+            clearTimeout(inDebounce);
+            inDebounce = setTimeout(func.bind(this, e), delay);
```

Comment:
func should not need bind because this.searchFunc has already been binded to this in the constructor.